Sources/NIOPerformanceTester/DeadlineNowBenchmark.swift

```diff
@@ -0,0 +1,39 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+final class DeadlineNowBenchmark: Benchmark {
+    private let iterations: Int
+
+    init(iterations: Int) {
+        self.iterations = iterations
+    }
+
+    func setUp() throws {
+    }
+
+    func tearDown() {
+    }
+
+    func run() -> Int {
+        var counter: UInt64 = 0
+        for _ in 0..<self.iterations {
+            let now = NIODeadline.now().uptimeNanoseconds
+            counter &+= now
+
+        }
+        return 1
```

Comment:
Can I suggest we return `counter` here? That way the benchmarks will be less inclined to kill the code.

Reply:
Sure, I think I saw another test returning `1` here. Do I understand correctly that this return value has no meaningful use, but is just used to ensure that code doesn't get optimised away?

Reply:
Addressed in https://github.com/apple/swift-nio/pull/2117/commits/edffb7d202d004c86c900ebe1bc85c7930482857.

Reply:
@simonjbeaumont Yes: the result is thrown away, but the call site doesn't know that.